src/audio-worklet/basic/hello-audio-worklet/main.js

```diff
@@ -3,24 +3,52 @@
 // found in the LICENSE file.
 
 const audioContext = new AudioContext();
+let oscillatorNode;
+let isPlaying = false;
 
 const startAudio = async (context) => {
   await context.audioWorklet.addModule('bypass-processor.js');
-  const oscillator = new OscillatorNode(context);
+  oscillatorNode = new OscillatorNode(context);
   const bypasser = new AudioWorkletNode(context, 'bypass-processor');
-  oscillator.connect(bypasser).connect(context.destination);
-  oscillator.start();
+  oscillatorNode.connect(bypasser).connect(context.destination);
+  oscillatorNode.start();
 };
 
 // A simplem onLoad handler. It also handles user gesture to unlock the audio
 // playback.
 window.addEventListener('load', async () => {
   const buttonEl = document.getElementById('button-start');
   buttonEl.disabled = false;
+
   buttonEl.addEventListener('click', async () => {
-    await startAudio(audioContext);
-    audioContext.resume();
-    buttonEl.disabled = true;
-    buttonEl.textContent = 'Playing...';
-  }, false);
+    if (!oscillatorNode) {
+      await startAudio(audioContext);
+      await audioContext.resume();
+      isPlaying = true;
+      buttonEl.textContent = 'Playing...';
+      buttonEl.classList.remove('start-button');
+    } else if (audioContext.state === 'suspended') {
+      await audioContext.resume();
+      isPlaying = true;
+      buttonEl.textContent = 'Playing...';
+      buttonEl.classList.remove('start-button');
+    } else {
+      await audioContext.suspend();
+      isPlaying = false;
+      buttonEl.textContent = 'START';
+      buttonEl.classList.add('start-button');
+    }
```

Comment:
Similarly to the previous comment, the logic between L24~L40 is slightly confusing.

I think it's possible to split the check of `oscillatorNode` out of this if/else block, so the block only handles the context state.